src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -630,16 +539,70 @@ public RuntimeExecutionResults Process()
             return instance;
         }
 
+        private bool AddDataSourcesCore(
+            IEnumerable<IDataSource> dataSources,
+            Type customDataSourceType,
+            List<CustomDataSourceReference> customDataSourceReferences,
+            Dictionary<CustomDataSourceReference, List<List<IDataSource>>> dataSourcesToProcess,
+            Func<Type, Type, bool> typeIs,
+            out ExceptionDispatchInfo error)
```

Comment:
>out ExceptionDispatchInfo error [](start = 12, length = 31)

this seems a bit awkward. would it make sense to use something like ErrorInfo, or is that not flexible enough?